contributors/design-proposals/daemonset-create-first-rolling-update.md

```diff
@@ -0,0 +1,182 @@
+## DaemonSet create-first rolling update strategy
+
+DaemonSets currently support a "delete first, then add" rolling update
+strategy.
+
+This document presents a design for a new "add first, then delete" rolling
+update strategy, called `SurgingRollingUpdate`. This strategy will create new
+pods first, cleaning up the old pods once the new ones are running. As with
+Deployments, users will use a `maxSurge` parameter to control how many nodes
+are allowed to have >1 pod running during a rolling update.
+
+### Example use case
+
+For self-hosted Kubernetes DaemonSets are a natural fit for running the
+scheduler and controller-manager on master nodes. However, since DaemonSets are
+managed by the controller-manager, if `maxUnavailable` >= the number of masters
```

Comment:
For this case, can we add validation to avoid such case? or make `maxUnavailable` to be percentage, e.g. 10% of master; if `maxUnavailable` can not meet, raise an event to identify that the upgrade failed.